libs/wingc/src/type_check.rs

```diff
@@ -450,12 +450,60 @@ impl Subtype for Type {
 					parent_type.is_subtype_of(other)
 				})
 			}
-			(Self::Resource(res), Self::Interface(_)) => {
+			(Self::Resource(res), Self::Interface(iface)) => {
 				// If a resource implements the interface then it's a subtype of it (nominal typing)
-				res.implements.iter().any(|parent| {
+				let implements_iface = res.implements.iter().any(|parent| {
 					let parent_type: &Type = parent;
 					parent_type.is_subtype_of(other)
-				})
+				});
+
+				if implements_iface {
+					return true;
+				}
+
+				// To support flexible inflight closures, we say that any
+				// resource with an inflight method named "handle" is a subtype of
+				// any single-method interface with a matching "handle" method type.
```

Comment:
Unless we want to annotate these interfaces in some way (`@handler`) and then have this logic "anded" with the fact that such an interface is also correctly annotated.

Reply:
I think I'll leave this as a future improvement for now